### PR TITLE
Fix a handful of fine-grained incremental bugs

### PR DESCRIPTION
**Pull request by msullivan**
This fixes all of the bugs found while running fine-grained incremental on the S repo with every file changed trivially.

### CODE REVIEW

File: test-data/unit/fine-grained.test [hunk 1]
```diff
@@ -2635,3 +2635,83 @@ def a(f: Callable[[Arg(int, 'x')], int]) -> None: pass
 [builtins fixtures/dict.pyi]
 [out]
 ==
+
+[case testImplicitTuple1]
+import a
+[file a.py]
+# Bogus annotation in nested function masked because
```
Comment (JukkaL) [inline test-data/unit/fine-grained.test]: Something missing in the comment? "... function masked because [what]"

File: mypy/server/update.py [hunk 2]
```diff
@@ -226,6 +228,11 @@ def update(self, changed_modules: List[Tuple[str, str]]) -> List[str]:
             changed_modules = [(id, path) for id, path in changed_modules
                                if id != next_id]
             changed_modules = dedupe_modules(remaining + changed_modules)
+            t1 = time.time()
+
+            self.manager.log_fine_grained(
```
Comment (JukkaL) [inline mypy/server/update.py]: This seems unrelated to the rest of thePR?

Comment (msullivan) [reply mypy/server/update.py]: Oh, indeed. I added it to track progress while running mypy-daemon on a big changed repo. I can split it out.
